experanto/data.py

```diff
@@ -30,3 +32,49 @@ def __getitem__(self, idx):
         times = self._sample_times[s : s + self.chunk_size]
         data, _ = self._experiment.interpolate(times)
         return self.DataPoint(*list(data.values()))
+
+
+class Mouse2pStaticImageDataset(Dataset):
+    def __init__(self, root_folder: str, tier: str, offset=100) -> None:
+        self.root_folder = Path(root_folder)
+        self.tier = tier
+        self.offset = offset
+        self._experiment = Experiment(root_folder)
+        self.device_names = self._experiment.device_names
+        self.DataPoint = namedtuple("DataPoint", self.device_names)
+        self._read_trials()
+    
+    def _read_trials(self):
+        screen = self._experiment.devices["screen"]
+        self._trials = [
+            t for t in screen.trials
+            if isinstance(t, ImageTrial) and t.get_meta("tier") == self.tier
+        ]
+        s_idx = np.array([t.first_frame_idx for t in self._trials])
+        if len(s_idx):
+            self._start_times = screen.timestamps[s_idx]
+            end_times = screen.timestamps[s_idx + 1]
+            self._trial_duration = np.min(end_times - self._start_times)
```

Comment:
@aecker I feel a bit uneasy about setting the trial duration to `np.min(end_times - self._start_times)`.

The minimum time could maybe be a few shorter than average. If you use that as the averaging window for all images, that could lead to problems.

I like though that it's data driven and does not rely on the meta info. For the images, we have the "stim duration=0.5" stored in each image meta file. 

I had a similar problem with adding the very last blank. Because the very last flip time does not exist, I just added the most common difference in time between two frames. But that's probably overkill. 
Looks like that:
```
digits = 4
most_common_time_diff_between_frames = np.unique(np.round(np.diff(all_flip_times), digits),
                    return_counts=True)[0][0]
```
the unique is returning as element [0][0] the most common occurence.

Reply:
see comment below for my change of heart